scripts/shift-player.js

```
@@ -214,7 +224,111 @@ var CastPlayer = function(element, options) {
+
+    try {
+      // Initialize the ads manager. Ad rules playlist will start at this time.
+      me.adsManager.init(me.element.clientWidth, me.element.clientHeight, google.ima.ViewMode.NORMAL);
+      // Call start to show ads. Single video and overlay ads will
+      // start at this time; this call will be ignored for ad rules, as ad rules
+      // ads start when the adsManager is initialized.
+      me.adsManager.start();
+    } catch (adError) {
+      console.log("adError", adError);
+      // An error may be thrown if there was a problem with the VAST response.
+    }
+  }
+
+  function contentEndedListener() {
+    me.adsLoader.contentComplete();
+  }
+
+  function onContentPauseRequested() {
+    // prevent putting the player into the pause state, when running the pre-roll
+    if (me.localPlayerState == PLAYER_STATE.IDLE)
+      return;
+
+    // This function is where you should setup UI for showing ads (e.g.
+    // display ad timer countdown, disable seeking, etc.)
+    me.localPlayer.removeEventListener('ended', me.contentEndedListener.bind(me));
+    me.pauseMedia();
+
+    if (me.adContainer)
+      me.adContainer.show();
+  }
+
+  function onContentResumeRequested() {
+    // This function is where you should ensure that your UI is ready to play content.
+    me.localPlayer.addEventListener('ended', me.contentEndedListener.bind(me));
+    me.playMedia();
+
+    if (me.adContainer)
+      me.adContainer.hide();
+  }
+
+  function onAdError(adErrorEvent) {
+    // Handle the error logging and destroy the AdsManager
+    console.log(adErrorEvent.getError());
+    if (me.adsManager){
+      me.adsManager.destroy();
+    }
+  }
+};
+
+CastPlayer.prototype.initializeAds = function(videoElement, adElement, options, width, height){
+  try
+  {
+    var adDisplayContainer = new google.ima.AdDisplayContainer(adElement[0], videoElement[0]);
```

I would personally prefer if you checked for the existance of `google.ima.AdDisplayContainer` instead of just going for it and swallowing the exception.